src/components/Navigation/_variables.scss

```diff
@@ -41,32 +42,36 @@ $nav-animation-variables: (
   text-decoration: none;
   text-align: left;
 
+  .Navigation-newDesignLanguage & {
+    font-weight: 600;
+    line-height: $item-line-height-large;
+    padding-left: spacing();
+  }
+
   &:hover {
-    background-image: var(
-      --p-override-none,
-      linear-gradient(color(state, hover), color(state, hover))
-    );
+    background-image: linear-gradient(color(state, hover), color(state, hover));
     background-color: var(--p-surface-hovered);
-    color: var(--p-action-primary, color('indigo', 'dark'));
+    color: var(--p-text, color('indigo', 'dark'));
     text-decoration: none;
 
     .Icon {
       @include recolor-icon(
-        var(--p-action-primary, color('indigo')),
+        var(--p-icon, color('indigo')),
         var(--p-override-transparent, color('white')),
         filter('indigo')
       );
     }
+
+    .Navigation-newDesignLanguage & {
+      background-image: var(--p-override-none);
+    }
   }
 
   @include focus-ring;
 
   &:focus {
     @include focus-ring($style: 'focused');
-    background-image: var(
-      --p-override-none,
-      linear-gradient(color(state, hover), color(state, hover))
-    );
```

Comment:
Good catch. I wonder if this was causing the same Safari problems in production and we just never noticed...